mongodump/mongodump_test.go

```diff
@@ -1186,7 +1213,9 @@ func TestMongoDumpViews(t *testing.T) {
 		err := setUpMongoDumpTestData()
 		So(err, ShouldBeNil)
 
-		err = setUpDBView()
+		colName := "dump_views"
+		dbName := testDB
+		err = setUpDBView(colName, dbName)
```

Comment:
I see some test refactoring in here, but I can't tell if there's anything in here that is actually regression testing the nil pointer bug

Reply:
I was testing another behavior of this bug that we don't hint index for mmapv1 cause we always consider them as views. TestMongoDumpTOOLS1952 has already covered this but we don't have mmapv1 set in evergreen. So I don't need to modify the test after adding mmapv1 to evergreen. 
For the Nil pointer error, I added test TestMongoDumpTOOLS2498 to reproduce it.